File: common/src/main/java/com/hartwig/actin/clinical/datamodel/Gender.java
package com.hartwig.actin.clinical.datamodel;

import com.hartwig.actin.Displayable;

import org.jetbrains.annotations.NotNull;

public enum Gender implements Displayable {
    MALE("Male"),
    FEMALE("Female");

    @NotNull
    private final String display;

    Gender(@NotNull final String display) {
        this.display = display;
    }

    @Override
    @NotNull
    public String display() {
        return display;
    }
}

File: datamodel/src/main/kotlin/com/hartwig/actin/datamodel/clinical/Gender.kt
package com.hartwig.actin.datamodel.clinical

import com.hartwig.actin.datamodel.Displayable

enum class Gender(private val display: String) : Displayable {
    MALE("Male"),
    FEMALE("Female"),
    UNKNOWN("");

    override fun display(): String {
        return display
    }
}
